src/main/java/sparksoniq/jsoniq/runtime/iterator/RuntimeIterator.java

```diff
@@ -191,13 +198,37 @@ public IteratorMetadata getMetadata() {
         return metadata;
     }
 
-    public abstract boolean isRDD();
+    public boolean isRDD() {
+        if (!isRDDIsSet) {
+            throw new RuntimeException("isRDD field accessed in iterator without being set.");
+        }
+        return isRDD;
```

Comment:
These exceptions are put in place as safeguards to prevent unnoticed bugs. They can be potentially be removed after the implementation is checked to be sound.